Lib/xml/dom/minidom.py

```diff
@@ -300,12 +300,28 @@ def _in_document(node):
         node = node.parentNode
     return False
 
-def _write_data(writer, data):
+def _write_data(writer, text, attr):
     "Writes datachars to writer."
-    if data:
-        data = data.replace("&", "&amp;").replace("<", "&lt;"). \
-                    replace("\"", "&quot;").replace(">", "&gt;")
-        writer.write(data)
+    if not text:
+        return
+    # See the comments in ElementTree.py for behavior and
+    # implementation details.
+    if "&" in text:
+        text = text.replace("&", "&amp;")
+    if "<" in text:
+        text = text.replace("<", "&lt;")
+    if ">" in text:
+        text = text.replace(">", "&gt;")
+    if attr:
+        if '"' in text:
+            text = text.replace('"', "&quot;")
+        if "\r" in text:
+            text = text.replace("\r", "&#13;")
+        if "\n" in text:
+            text = text.replace("\n", "&#10;")
+        if "\t" in text:
+            text = text.replace("\t", "&#09;")
```

Comment:
So…
* saxutils: `&#9;`
* ElementTree: `&#09;`
* lxml: `&#9;`
* C14N `&#x9;`
* minidom: `\t`

What a lovely mess. Since minidom is the only code we're touching here, should we just go for the shortest form `&#9;`? It matches what saxutils and lxml output, although not what ElementTree does. But if we ever decide to change other libraries here (for which there's currently no reason whatsoever), it's probably easiest to arbitrarily agree on the shortest form in the face of a total lack of other arguments.

```suggestion
            text = text.replace("\t", "&#9;")
```